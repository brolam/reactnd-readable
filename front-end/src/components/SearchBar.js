import React from 'react'
import { DebounceInput } from 'react-debounce-input';

function SearchBar({
  placeholder,
  onSearch = value => { console.log(value) } }) {
  return (
    <div className="search-bar">
      <DebounceInput
        placeholder={placeholder}
        minLength={2}
        debounceTimeout={300}
        onChange={event => onSearch(event.target.value)}
      />
    </div>
  )
}

export default SearchBar