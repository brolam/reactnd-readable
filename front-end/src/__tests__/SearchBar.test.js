import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import SearchBar from '../components/SearchBar'

test('render without error', () => {
  ReactDOM.render(<SearchBar placeholder="Search by title post" />, document.createElement('div'));
})

test('last Snapshot', () => {
  const searchBar = renderer.create(
    <SearchBar
      placeholder="Search by title post"
    />
  );
  let tree = searchBar.toJSON();
  expect(tree).toMatchSnapshot();
})

test('debounce input onSearch', () => {
  jest.useFakeTimers();
  let searchValue = undefined;
  const onSearch = (value) => {
    searchValue = value
  }
  const searchBar = mount(
    <SearchBar
      placeholder="Search by title post"
      onSearch={onSearch}
    />
  );
  const searchBarInput = searchBar.find('input')
  const event = { target: { value: 'udacity' } };
  searchBarInput.simulate('change', event);
  expect(searchValue).toBe(undefined)
  setTimeout(() => {
    expect(searchValue).toBe('udacity')
  }, 3000);
  jest.runAllTimers();
})

