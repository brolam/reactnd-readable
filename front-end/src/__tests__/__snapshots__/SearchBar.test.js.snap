// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`last Snapshot 1`] = `
<div
  className="search-bar"
>
  <input
    onBlur={[Function]}
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Search by title post"
    type="text"
    value=""
  />
</div>
`;
